### ISSUE TITLE
Compare Swagger code gen and openapi code gen

### ISSUE THREAD

**Issue by tschaffter** [nlpsandbox/date-annotator-example#50]
- https://github.com/swagger-api/swagger-codegen
- https://github.com/OpenAPITools/openapi-generator

**Comment by tschaffter** [comment]
## Observations

- The generator says for the test files that they are "Skipped by user option(s)" but I have not specified this option.
- There are errors shown. The s[ame errors are visible in the official documentation of the generator](https://openapi-generator.tech/docs/debugging/) so is it normal?

```
$ openapi-generator generate -g python-flask -o server --dry-run -i $npm_config_spec
> awesome-date-annotator@0.2.1 generate:server /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example
> openapi-generator generate -g python-flask -o server --dry-run -i $npm_config_spec "dist.yaml"

[main] INFO  o.o.codegen.DefaultGenerator - Generating with dryRun=true
[main] INFO  o.o.codegen.DefaultGenerator - OpenAPI Generator: python-flask (server)
[main] INFO  o.o.codegen.DefaultGenerator - Generator 'python-flask' is considered stable.
[main] INFO  o.o.c.l.AbstractPythonConnexionServerCodegen - Environment variable PYTHON_POST_PROCESS_FILE not defined so the Python code may not be properly formatted. To define it, try 'export PYTHON_POST_PROCESS_FILE="/usr/local/bin/yapf -i"' (Linux/Mac)
[main] INFO  o.o.c.l.AbstractPythonConnexionServerCodegen - NOTE: To enable file post-processing, 'enablePostProcessFile' must be set to `true` (--enable-post-process-file for CLI).
[main] ERROR o.o.codegen.DefaultGenerator -

Dry Run Results:

i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/.dockerignore
k /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/.openapi-generator-ignore
w /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/.openapi-generator/VERSION
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/.travis.yml
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/Dockerfile
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/git_push.sh
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/openapi_server/__main__.py
k /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/openapi_server/test/test_service_controller.py
k /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/openapi_server/test/test_text_date_annotation_controller.py
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/requirements.txt
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/setup.py
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/test-requirements.txt
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/tox.ini


States:

  - w Write
  - n Write if New/Updated
  - i Ignored
  - s Skipped Overwrite
  - k Skipped by user option(s)
  - e Error evaluating file write state


[main] ERROR o.o.codegen.DefaultGenerator -

Dry Run Results:

i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/.dockerignore
k /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/.openapi-generator-ignore
w /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/.openapi-generator/VERSION
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/.travis.yml
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/Dockerfile
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/git_push.sh
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/openapi_server/__main__.py
k /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/openapi_server/test/test_service_controller.py
k /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/openapi_server/test/test_text_date_annotation_controller.py
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/requirements.txt
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/setup.py
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/test-requirements.txt
i /mnt/c/Users/thoma/Documents/dev/nlpsandbox/date-annotator-example/server/tox.ini


States:

  - w Write
  - n Write if New/Updated
  - i Ignored
  - s Skipped Overwrite
  - k Skipped by user option(s)
  - e Error evaluating file write state


```

**Comment by tschaffter** [comment]
>Test files are NOT overwritten as part of the code generation by default as someone may have updated the test files with some manually-written test cases already.

Source: https://github.com/OpenAPITools/openapi-generator/issues/4075#issuecomment-539528249

This behavior is not intuitive, given that controller files are overwritten and obviously they have been edited too.

**Comment by tschaffter** [comment]
Early in the development of this project, we first tried swagger code gen but faced non-trivial errors. Using openapi-generator instead solved the problem and we didn't have any issues so far when generating server and client code in python and java.

Openapi-generator is developed by a communty while swagger code-gen is developed by a company. Openapi-generator has been forked from swagger code gen and is supported by developers who were initially working on swagger code gen. I looked at the history of contributions for both projects and openapi-generator development is moving faster than swagger code gen.
